src/test/shell/integration/exec_group_test.sh

```diff
@@ -297,6 +297,71 @@ EOF
   grep "test_value" out.txt || fail "Did not find the test-action value"
 }
 
+function test_platform_execgroup_properties_starlark_test() {
+  local -r pkg=${FUNCNAME[0]}
+  mkdir $pkg || fail "mkdir $pkg"
+
+  if "$is_windows"; then
+    script_name="test_script.bat"
+    script_content="@echo off\necho hello\n"
+  else
+    script_name="test_script.sh"
+    script_content="#!/bin/bash\necho hello\n"
+  fi
+
+  cat > ${pkg}/defs.bzl <<EOF
+def _impl(ctx):
+    out_file = ctx.actions.declare_file("$script_name")
+    ctx.actions.write(out_file, "$script_content", is_executable=True)
+    return [DefaultInfo(executable = out_file)]
+
+starlark_test = rule(
+    implementation = _impl,
+    test = True,
+)
+
+starlark_with_manual_test_exec_group_test = rule(
+    implementation = _impl,
+    test = True,
+    exec_groups = {
+        "test": exec_group(),
+    },
+)
+EOF
+  cat > ${pkg}/BUILD <<EOF
+load("//$pkg:defs.bzl", "starlark_test", "starlark_with_manual_test_exec_group_test")
+constraint_setting(name = "setting")
+constraint_value(name = "local", constraint_setting = ":setting")
+starlark_test(
+  name = "a",
+  exec_compatible_with = [":local"],
+)
+
+starlark_with_manual_test_exec_group_test(
+  name = "b",
+  exec_compatible_with = [":local"],
+)
+
+platform(
+    name = "my_platform",
+    parents = ["${default_host_platform}"],
+    exec_properties = {
+        "platform_key": "default_value",
+        "test.platform_key": "test_value",
+    },
+    constraint_values = [":local"],
+)
+EOF
+
+  bazel test --extra_execution_platforms="${pkg}:my_platform" ${pkg}:a --execution_log_json_file out.txt || fail "Test failed"
+  grep "platform_key" out.txt || fail "Did not find the platform key"
+  grep "test_value" out.txt || fail "Did not find the test-action value"
+
+  bazel test --extra_execution_platforms="${pkg}:my_platform" ${pkg}:b --execution_log_json_file out.txt || fail "Test failed"
```

Comment:
Can you split this into two separate tests? One for the default test exec group, one for the manual? I think that will make any future errors easier to diagnose.

Reply:
Done.